src/core/meshcop/timestamp.cpp

```diff
@@ -62,16 +58,30 @@ int Timestamp::Compare(const Timestamp *aFirst, const Timestamp *aSecond)
 
     // Both are non-null.
 
-    firstSeconds  = aFirst->GetSeconds();
-    secondSeconds = aSecond->GetSeconds();
+    rval = Compare(*aFirst, *aSecond);
+
+exit:
+    return rval;
+}
+
+int Timestamp::Compare(const Timestamp &aFirst, const Timestamp &aSecond)
```

Comment:
It seems to compare the two timestamps as big-endian uint64_t (without the authoritative bits). 

If so, the method can be simplified by comparing uint64_t values directly.

Reply:
There is no `uint64_t` value.

Reply:
Any way, I changed to use `uint64_t` as the underlying value. PTAL